### PR TITLE
Update Open Ephys events

### PR DESCRIPTION
**Pull request by alejoe91**
### Updates for v0.6:

In the new OE version, `timestamps.npy` are already in seconds and do not need scaling by the sampling frequency (https://open-ephys.github.io/gui-docs/User-Manual/Recording-data/Binary-format.html#events)

### Other updates:

In case the event stream has `states.npy`, the event channel type is an `epoch` since the duration can be computed using positive and negative states for each TTL channel.

### CODE REVIEW

**Comment by JuliaSprenger** [comment]
Hi @alejoe91 Thanks for the update. Unfortunately it seems like your changes are messing up the array_annotations of old style datasets. Could you have a look at this?
Also we can add a dataset in the new format to gin for testing. Do you have a small one available?

**Comment by alejoe91** [comment]
@JuliaSprenger I think I fixed it :) however, another core test seems to be failing..can you take a look?

File: neo/rawio/openephysbinaryrawio.py [hunk 1]
```diff
@@ -204,6 +204,32 @@ def _parse_header(self):
...
 
+                    # # If available, use 'states' to compute event duration
+                    if 'states' in d:
+                        states = d["states"]
+                        timestamps = d["timestamps"]
+                        labels = d["labels"]
+                        rising = np.where(states > 0)[0]
+                        falling = np.where(states < 0)[0]
+                        # make sure first event is rising and last is falling
+                        if states[0] < 0:
```
Comment (JuliaSprenger) [inline neo/rawio/openephysbinaryrawio.py:215]: What about the case where states == 0 or multiple values in the beginning or end are smaller or bigger than 0 respectively?

**Comment by JuliaSprenger** [comment]
Tests are passing now, the failure was one of those rare spiketrainlist errors, see #1118

File: neo/rawio/openephysbinaryrawio.py (hunk 1 above)
Comment (alejoe91) [reply neo/rawio/openephysbinaryrawio.py:215]: I thin that `states` are by definition either `+channel_number` or `-channel_number`.

@jsiegle could you confirm that?

**Comment by alejoe91** [comment]
> Hi @alejoe91 Thanks for the update. Unfortunately it seems like your changes are messing up the array_annotations of old style datasets. Could you have a look at this? Also we can add a dataset in the new format to gin for testing. Do you have a small one available?

The `v0.6.x_neuropixels_multiexp_multistream` is already from the new version. You now have `sample_numbers.npy` (samples) and `timestamps.npy` (seconds).

File: neo/rawio/openephysbinaryrawio.py (hunk 1 above)
Comment (jsiegle) [reply neo/rawio/openephysbinaryrawio.py:215]: That's right, `states` can never be zero. However, it is possible that first event can be falling, e.g. if a TTL line was high when the recording started. Same with the last event being rising.

Comment (alejoe91) [reply neo/rawio/openephysbinaryrawio.py:215]: That's taken care of :) in that case the first/last event is dropped

**Comment by JuliaSprenger** [comment]
> > Hi @alejoe91 Thanks for the update. Unfortunately it seems like your changes are messing up the array_annotations of old style datasets. Could you have a look at this? Also we can add a dataset in the new format to gin for testing. Do you have a small one available?
> 
> The `v0.6.x_neuropixels_multiexp_multistream` is already from the new version. You now have `sample_numbers.npy` (samples) and `timestamps.npy` (seconds).

I don't think this session is included in tests yet. Could you add it to also ensure the new format is read correctly?

**Comment by alejoe91** [comment]
It is tested already! Both for [rawio](https://github.com/alejoe91/python-neo/blob/update_oe_events/neo/test/rawiotest/test_openephysbinaryrawio.py#L16) and [iotests](https://github.com/alejoe91/python-neo/blob/97509e6c11a9a56cda9ac11786f99c4019c265b6/neo/test/rawiotest/test_openephysbinaryrawio.py#L16)

**Comment by JuliaSprenger** [comment]
Sorry, I missed that reference. Then this is complete for me.
Merging now :)

**Pull request merged**